src/main/capi/arrow-c.cpp

```diff
@@ -108,3 +108,112 @@ duckdb_state duckdb_execute_prepared_arrow(duckdb_prepared_statement prepared_st
 	*out_result = (duckdb_arrow)arrow_wrapper;
 	return !arrow_wrapper->result->HasError() ? DuckDBSuccess : DuckDBError;
 }
+
+namespace arrow_array_stream_wrapper {
+namespace {
+struct PrivateData {
+	ArrowSchema *schema;
+	ArrowArray *array;
+	bool done;
+};
+
+void EmptySchemaRelease(ArrowSchema *schema) {
+}
+
+void GetSchemaVoid(uintptr_t stream_factory_ptr, duckdb::ArrowSchemaWrapper &schema) {
+	auto private_data = (PrivateData *)((ArrowArrayStream *)stream_factory_ptr)->private_data;
+	schema.arrow_schema = *private_data->schema;
+	schema.arrow_schema.release = EmptySchemaRelease;
+
+	for (int64_t i = 0; i < schema.arrow_schema.n_children; i++) {
+		schema.arrow_schema.children[i]->release = EmptySchemaRelease;
+	}
```

Comment:
I think the nullification of the release would be cleaner at the `duckdb_arrow_scan` function